jax_unirep/utils.py

```diff
@@ -310,6 +311,32 @@ def batch_sequences(seqs: List[str]) -> List[List]:
     return order
 
 
+def right_pad(seqs: List[str], max_len: int):
+    """Pad all seqs in a list to longest length on the right with "-"."""
+    return [seq.ljust(max_len, "-") for seq in seqs]
+
+
+def get_random_batch(seqs: List[str], batch_size: int = 25):
+    """
+    Batch random sequences, pad all sequences to longest length seq in batch.
+
+    This function randomly samples `batch_size` sequences and
+    applies padding to make all sequences in the batch same-length.
+
+    :param seqs: List of sequences as strings.
+    :param batch_size: Batching size
+    :returns: One batch of sequences
+    """
+
+    if batch_size > len(seqs):
+        raise ValueError(
+            "Batch size cannot be bigger than total number of sequences."
+        )
+    subset = sample(seqs, batch_size)
+    max_len = max([len(seq) for seq in subset])
+    return right_pad(subset, max_len)
```

Comment:
I'm thinking this should be changed to help make random batching a subset of length batching.

Rather than return a sample of sequences, return a sample of array indices.

Straw man implementation w/o docstrings (don't be lazy, add in the docstrings to your best understanding! 👅 ):

```python
def get_random_batch(xs: np.ndarray, ys: np.ndarray, batch_size: int = 25):
    idxs = np.random.choice(list(range(len(xs))), batch_size)
    return xs[idxs], ys[idxs]
```

That's the gist of it.